src/Extensions/Hosting.Services.Web/Middlewares/UserIdentity/UserIdentiyMiddleware.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Buffers;
+using System.Diagnostics;
+using System.Security.Cryptography;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Omex.Extensions.Abstractions.Activities;
+
+namespace Microsoft.Omex.Extensions.Hosting.Services.Web.Middlewares
+{
+	/// <summary>
+	/// Adds user information if needed
+	/// </summary>
+	internal class UserIdentiyMiddleware : IMiddleware, IDisposable
```

Comment:
Consider renaming to `UserHashIdentityMiddleware` or something similar

Reply:
Renamed